src/com/google/javascript/jscomp/parsing/parser/Parser.java

```diff
@@ -2824,7 +2824,7 @@ private ParseTree parseAssignment(Expression expressionIn) {
 
     if (isStartOfAsyncArrowFunction(left)) {
       // re-evaluate as an async arrow function.
-      resetScanner(left);
+      resetScanner(start);
```

Comment:
The intention here is "reparse `left` as something else".
Resetting to an arbitrary position isn't something I think we should be doing.